crengine/src/lvxml.cpp

```diff
@@ -2989,8 +2989,10 @@ bool LVXMLParser::Parse()
                         m_state = ps_text;
                         break;
                     }
-                    //bypass <![CDATA] in <style type="text/css">
-                    if (PeekCharFromBuffer(1)=='['&&tagname.compare("style")==0&&attrvalue.compare("text/css")==0){
+                    // bypass <![CDATA] in <style type="text/css"> (and more generally in any <style>)
+                    // (We let the trailing ]]> in, our style parser will just ignore it)
+                    // todo: ignoring CDATA should be handled more generally and properly
+                    if ( PeekCharFromBuffer(1)=='[' && tagname.compare("style")==0 ) { // && attrvalue.compare("text/css")==0){
```

Comment:
As I wrote in https://github.com/koreader/crengine/issues/396#issuecomment-737090664 perhaps it should be this instead (but I haven't checked what browsers do to deal with real life documents in the meantime):
```suggestion
                    if ( PeekCharFromBuffer(1)=='[' && tagname.compare("style")==0 ) || attrvalue.compare("text/css")==0) {
```

Reply:
Not sure I understand.
What else do you want to catch ? This seems (with some added parens) it would catch `<foo bar="text/css"><[[CDATA...` ?
(tagname is the current tag we're in, and attrvalue is the last attribute value met)

Reply:
We would anyway parse any `<style>` or `<style type=blah>` as CSS, if they do not start with `<[[CDATA`.
And previously, if they did, we would just skip content up to the `>` in `P > * {color: blue;}` - which is even more disastrous as the used stylesheet would then be `* {color: blue;}` :/

Reply:
> What else do you want to catch ?

I don't know if I want anything at all, that depends on what browsers do. But per HTML5, only `<style>` and `<style type="text/css">` should be regarded as CSS, while anything else should be discarded as unsupported.

Reply:
Well, then that should be handled at a later step, when we check for a `style` tag to parse and apply its text content as a stylesheet. If no type attribute, or type == "text/css": then apply it. If type and type != text/css, just don't parse and don't apply it.
At this early step of XML parsing, we should parse it correctly (which we don't :) and include it in the DOM.

Reply:
> At this early step of XML parsing

I was missing that part before. It has no place in that. :-)